python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -436,6 +438,10 @@ def _executeQuantaInProcess(self, graph, butler):
                 self.quantumExecutor.execute(qnode.taskDef, qnode.quantum, butler)
                 successCount += 1
             except Exception as exc:
+                if self.pdb and self.numProc == 1:
+                    import pdb  # noqa E402: only load this when we need it
```

Comment:
Ooooh, while we're in here, any chance of changing this to be something I've wanted for ages?!

```
    try:
        import ipdb as pdb:
    except ImportError:
        import pdb
```

Reply:
What if `ipdb` is available, but I actually want `pdb`?
I think this would need some more configuration.

Reply:
Oh, I guess technically that's possible, I admit I didn't think of that. I assumed that `ipdb` was pretty much a superset of pdb, and that if someone had gone to the trouble of installing it, that would mean it was their debugger of choice, but maybe neither of those assumptions are actually true, and sometimes you actually need strictly `pdb`.

Reply:
OK, changed to allow `--pdb=ipdb`.

Reply:
Oh amazing, thank you!

Reply:
Could we get open-ended debugger choice by just using `breakpoint()`, and trying the user to run inside their favorite debugger?  I believe that's what `breakpoint()` is for.

Reply:
`breakpoint()` is for getting a debugger at the current point of execution. However, we want to get a debugger at the point of execution where the exception was raised. It's the difference between `pdb.set_trace()` and `pdb.post_mortem()`.
We could have Merlin set `PYTHONBREAKPOINT=ipdb.post_mortem`, but I don't think that would work if he's sprinkled other `breakpoint()` calls throughout his code.

Reply:
I don't think I'd tend to mix-and-match other breakpoints with this, but who knows. Also though, I know that both pdb and ipdb support post_mortem, so I'm not sure what other debuggers there are which wouldn't work with this. Basically, I think this is great as it is 🙂